python/ray/dashboard/utils.py

```diff
@@ -668,8 +668,15 @@ def ray_address_to_api_server_url(address: Optional[str]) -> str:
     Returns:
         API server HTTP URL.
     """
+    # If no address is provided and this worker is connected to a Ray instance,
+    # default to that Ray instance's address.
+    if address is None:
+        worker = ray._private.worker.global_worker
+        if worker.connected:
+            address = worker.node.address_info["address"]
```

Comment:
I couldn't find an existing less hacky way to do this and it seems to be the pattern we follow throughout the code. LMK if there's a better suggestion.